### ISSUE TITLE
entity_load_multiple requires ids, even though it's marked optional in the documentation

### ISSUE THREAD

**Issue by nathanharper** [drupal-graphql/graphql#977]
I think the title says it all here. I think this is a bug, right? I'd think just loading by type and bundle would be a common use case

**Comment by joaogarin** [comment]
Hello, In documentation you mean the annotation in https://github.com/drupal-graphql/graphql/blob/8.x-4.x/src/Plugin/GraphQL/DataProducer/Entity/EntityLoadMultiple.php#L30 ?

**Comment by nathanharper** [comment]
Yes, it seems like _not_ specifying "required" in the annotation defaults it to TRUE, which is inconsistent with the documentation, which says it is optional: https://drupal-graphql.gitbook.io/graphql/v/8.x-4.x/data-producers/built-in#entity-data-producers

**Comment by Kingdutch** [comment]
I ran into the same issue. I think changing the documentation may be the best course of action:

- Adding `required = FALSE` to the annotation requires a function signature change for `EntityLoadMultiple::resolve`
- Changing the signature to allow NULL will work, however it doesn't have the expected outcome (load all entities) but instead loads nothing, so this would probably create more confusion than documenting that IDs are required.

Even if we managed to fix the behaviour for the resolver to load all entities when `NULL` is provided, this is dangerous to put in new users' happy path, since accidentally loading 10.000 entities could bring down the server. Instead an alternative should be implemented in user-code (or possibly provided by default) that allows paged loading and has a default limit. 

The right route is a bit more work but ensures GraphQL APIs built using this module will be robust :)

**Comment by edurenye** [comment]
Agreed with @Kingdutch, and also would be great to have an example of 'entity_load_multiple' in the examples.
